test/framework/event-filters.ts

```diff
@@ -0,0 +1,225 @@
+import {ContractReceipt, utils} from 'ethers'
+import {Log} from '@ethersproject/abstract-provider'
+import {expect} from 'chai'
+import {TypedEvent, TypedEventFilter} from '../../typechain-types/common'
+
+type DecodeFunc = (log: Log) => utils.Result
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export interface ExtendedEventFilter<T = any>
```

Comment:
question: is there a way of typing T?

_While Typing is valuable, as it's still binding to JS, it's not always possible._

Reply:
well, there is no actual type here ... but can put `object` here instead of `any`

Reply:
...well, that would avoid the problem of using `any` 😁